File: OpenAI/assistent/assistent.py
import requests
import time

# Replace these variables with your actual endpoint and API key
endpoint = '<<your instance>>.openai.azure.com'
api_key = '<<your key>>'

url = f"https://{endpoint}/openai/assistants?api-version=2024-05-01-preview"
headers = {
    'api-key': api_key,
    'Content-Type': 'application/json',
}
data = {
    "instructions": "python code creator dsafdasf",
      "name": "testrb",
  "tools": [],
  "model": "gpt-4",
  "tool_resources": {
    "code_interpreter": {
      "file_ids": [
        "assistant-s8grUFK2VIT08OyTfvXdFdJG",
      ]
    }
  },
  "temperature": 1,
  "top_p": 1
}

response = requests.post(url, headers=headers, json=data)
assistent_id = response.json().get("id")
print(assistent_id)

# Create a thread
create_thread_url = f"https://{endpoint}/openai/threads?api-version=2024-05-01-preview"
create_thread_response = requests.post(create_thread_url, headers=headers, data='{}')
thread_id = create_thread_response.json().get("id")  # Assuming the response contains an "id" field
print(thread_id)

# Add a user question to the thread
add_question_url = f"https://{endpoint}/openai/threads/{thread_id}/messages?api-version=2024-05-01-preview"
question_data = {
  "role": "user",
  "content": "give me python code to store data on adlsgen2"
}
add_question_response = requests.post(add_question_url, headers=headers, json=question_data)

# Run the thread
run_thread_url = f"https://{endpoint}/openai/threads/{thread_id}/runs?api-version=2024-05-01-preview"
run_data = {
    "assistant_id": assistent_id,  # Assuming this is a string, it should be quoted
}

run_thread_response = requests.post(run_thread_url, headers=headers, json=run_data)
run_id = run_thread_response.json().get("id")
print(run_id)

run_id_url = f"https://{endpoint}/openai/threads/{thread_id}/runs/{run_id}?api-version=2024-05-01-preview"
run_id_status = requests.get(run_id_url, headers=headers)
#print(run_id_status.json())

i=1
while run_id_status.json().get("status") in ['queued', 'in_progress', 'cancelling']:
    time.sleep(1)
    i= i + 1
    run_id_url = f"https://{endpoint}/openai/threads/{thread_id}/runs/{run_id}?api-version=2024-05-01-preview"
    run_id_status = requests.get(run_id_url, headers=headers)
    print(str(i) + ", " + str(run_id_status.json().get("status")))

response_thread_url = f"https://{endpoint}/openai/threads/{thread_id}/messages?api-version=2024-05-01-preview"
response_response= requests.get(response_thread_url, headers=headers)

print(response_response.json())

File: OpenAI/assistent/assistent.java
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;

public class OpenAIRequestJava {
    private static final String endpoint = "ai-rebremerai744852064041.openai.azure.com";
    private static final String apiKey = "995cdb2a8e6b4fa18d739489fe001ca3";
    private static final HttpClient client = HttpClient.newBuilder()
            .version(HttpClient.Version.HTTP_2)
            .connectTimeout(Duration.ofSeconds(10))
            .build();

    public static void main(String[] args) throws Exception {
        // Create Assistant
        String assistantId = createAssistant();
        System.out.println(assistantId);

        // Create a Thread
        String threadId = createThread();
        System.out.println(threadId);

        // Add a User Question to the Thread
        addUserQuestion(threadId);

        // Run the Thread
        String runId = runThread(threadId, assistantId);
        System.out.println(runId);

        // Check Run Status
        checkRunStatus(threadId, runId);

        // Get Thread Messages
        getThreadMessages(threadId);
    }

    private static String createAssistant() throws Exception {
        String url = String.format("https://%s/openai/assistants?api-version=2024-05-01-preview", endpoint);
        String requestBody = "{\"instructions\": \"python code creator dsafdasf\", \"name\": \"testrb\", \"tools\": [], \"model\": \"gpt-4\", \"tool_resources\": {\"code_interpreter\": {\"file_ids\": [\"assistant-s8grUFK2VIT08OyTfvXdFdJG\"]}}, \"temperature\": 1, \"top_p\": 1}";

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("Authorization", "Bearer " + apiKey)
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        JsonObject jsonResponse = JsonParser.parseString(response.body()).getAsJsonObject();
        return jsonResponse.get("id").getAsString();
    }

    private static String createThread() throws Exception {
        String url = String.format("https://%s/openai/threads?api-version=2024-05-01-preview", endpoint);

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("Authorization", "Bearer " + apiKey)
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString("{}"))
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        JsonObject jsonResponse = JsonParser.parseString(response.body()).getAsJsonObject();
        return jsonResponse.get("id").getAsString();
    }

    private static void addUserQuestion(String threadId) throws Exception {
        String url = String.format("https://%s/openai/threads/%s/messages?api-version=2024-05-01-preview", endpoint, threadId);
        String requestBody = "{\"role\": \"user\", \"content\": \"give me python code to store data on adlsgen2\"}";

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("Authorization", "Bearer " + apiKey)
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
                .build();

        client.send(request, HttpResponse.BodyHandlers.ofString());
    }

    private static String runThread(String threadId, String assistantId) throws Exception {
        String url = String.format("https://%s/openai/threads/%s/runs?api-version=2024-05-01-preview", endpoint, threadId);
        String requestBody = String.format("{\"assistant_id\": \"%s\"}", assistantId);

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("Authorization", "Bearer " + apiKey)
                .header("Content-Type", "application/json")
                .POST(HttpRequest.BodyPublishers.ofString(requestBody))
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        JsonObject jsonResponse = JsonParser.parseString(response.body()).getAsJsonObject();
        return jsonResponse.get("id").getAsString();
    }

    private static void checkRunStatus(String threadId, String runId) throws Exception {
        String url;
        HttpResponse<String> response;
        JsonObject jsonResponse;
        String status;

        do {
            url = String.format("https://%s/openai/threads/%s/runs/%s?api-version=2024-05-01-preview", endpoint, threadId, runId);
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create(url))
                    .header("Authorization", "Bearer " + apiKey)
                    .build();

            response = client.send(request, HttpResponse.BodyHandlers.ofString());
            jsonResponse = JsonParser.parseString(response.body()).getAsJsonObject();
            status = jsonResponse.get("status").getAsString();
            System.out.println(status);

            Thread.sleep(1000); // Sleep for 1 second
        } while ("queued".equals(status) || "in_progress".equals(status) || "cancelling".equals(status));
    }

    private static void getThreadMessages(String threadId) throws Exception {
        String url = String.format("https://%s/openai/threads/%s/messages?api-version=2024-05-01-preview", endpoint, threadId);

        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(url))
                .header("Authorization", "Bearer " + apiKey)
                .build();

        HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
        System.out.println(response.body());
    }
}
